[SRO] Create basic Stat Filter

tiffanynwyeung opened the pull request:
## Describe your changes
- create basic stat filter UI
- add and modify `StatFilterSetting` to now have `minValue` and `maxValue`
- add `StatFilterCard` to temp optimize page for viewing purposes

## Issue or discord link
* Resolve #2308 

## Testing/validation
Appearance when `targetSelector` on left, with checkbox/delete/input enabled:
![image](https://github.com/user-attachments/assets/53599eb8-3df0-4043-bda7-a3d803459f25)

Alternative layout with `targetSelector` in the middle, with input enabled and disabled respectively:
<img width="1183" alt="image" src="https://github.com/user-attachments/assets/54fb5eb4-069f-4e96-843a-c9dd1a008e40">
<img width="1174" alt="image" src="https://github.com/user-attachments/assets/a642a1e4-3078-47bd-8972-40c39cf4054a">


## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

nguyentvan7 reviewed (commented):
Looks good to me, we can refine later as we plan out the opt-tab anyways. I think only comment is maybe we should put the min filter on the left side of the stat select? or is that too clunky/interrupts the left-to-right flow?

tiffanynwyeung commented:
I've added the alternative layout with `targetSelector` in the center with `min` on the left. I think the flow may be a bit awkward with the checkbox on the left, and there may be potential accessibility issues with needing to select the `targetSelector` first before being able to change input in `min` and `max`? Outside of the checkbox on the left, the middle selector does look better with the current `buttonGroup` design.

I also am not too sure about how to set `maxValue` when there is no user input, from looking at how `StatFilter` in GO gets parsed. It's probably a "putting a cart before the horse" thought exercise, so I decided on setting it to `Infinity` for now, and the logic can be revised when properly implementing opt-tab later. I did come up with the following though: 
- current solution: set `max` to `Infinity`
- making `min` and `max` optional properties that can be removed and added
- possibly accepting `undefined` for `min` and `max` as the default 'no input' value, and later opt-tab code filters out `max/min = undefined`

nguyentvan7 reviewed (approved):
Looks good for now thanks!
